CST-to-AST/type-information.lisp

```diff
@@ -9,10 +9,14 @@
 ;;; repeatedly. That is to say, this is macroexpand, not
 ;;; macroexpand-1 or macroexpand-all.
 
-(defgeneric type-expand (environment type-specifier))
+(defgeneric type-expand (client type-specifier environment))
 
-(defmethod type-expand ((env entry) type-specifier)
-  (type-expand (next env) type-specifier))
+;; FIXME(paul) I don't think this expands repeatedly.
+;; Then again, what the other one used to do is call TYPE-EXPAND until
+;; the environment is NIL.
```

Comment:
Having a default method call the generic function again is a bit dangerous if the client forgets to specialize it. It might be better to have an ordinary `type-expand` function that calls the generic with the `trucler:global-environment` to avoid this.